File: LC383.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        map <char, int> m1;

        for(auto i : magazine)
            if(m1.count(i) == 0)
                m1[i] = 1;
            else
                m1[i]++;

        for(auto i : ransomNote)
            if(m1.count(i) != 0)
	      if(--m1[i] < 0)
		return false;
            else
                return false;
        
        return true;
    }
};

File: LC383.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        d = {}

        for i in magazine:
            if i in d:
                d[i] += 1
            else:
                d[i] = 1
        
        for i in ransomNote:
            if i in d:
                d[i] -= 1
                if d[i] < 0:
                    return False
            else:
                return False
        
        return True
